src/components/fabSpeedDial/fabSpeedDial.scss

```diff
@@ -0,0 +1,167 @@
+md-fab-speed-dial {
+  $custom-duration: $swift-ease-in-duration / 2;
+  $custom-transition: all $custom-duration $swift-ease-in-timing-function !default;
+
+  $trigger-width: 56px;
+  $trigger-height: 52px;
+  $trigger-offset: 8px;
+
+  display: flex;
+  align-items: center;
+
+  &.md-opened {
+    &.fling {
+      md-fab-actions {
+        .md-fab-action-item {
+          visibility: visible;
+          transform: translateY(0px) !important;
+        }
+      }
+    }
+
+    &.scale {
+      md-fab-actions {
+        .md-fab-action-item {
+          visibility: visible;
+          transform: scale(1) !important;
+          transition: $custom-transition;
+        }
+      }
+    }
+  }
+
+  md-fab-actions {
+    display: flex;
+
+    // Set the height so that the z-index below works properly
+    height: 100%;
+
+    .md-fab-action-item {
+      visibility: hidden;
+    }
+  }
+
+  &.md-down {
+    flex-direction: column;
+
+    md-fab-trigger {
+      order: 1;
+    }
+
+    md-fab-actions {
+      flex-direction: column;
+      order: 2;
+
+      .md-fab-action-item {
+        transition: $custom-transition;
+      }
+    }
+  }
+
+  &.md-up {
+    flex-direction: column;
+
+    md-fab-trigger {
+      order: 2;
+    }
+
+    md-fab-actions {
+      flex-direction: column-reverse;
+      order: 1;
+
+      .md-fab-action-item {
+        transition: $custom-transition;
+      }
+    }
+  }
+
+  &.md-left {
+    flex-direction: row;
+
+    md-fab-trigger {
+      order: 2;
+    }
+
+    md-fab-actions {
+      flex-direction: row-reverse;
+      order: 1;
+
+      .md-fab-action-item {
+        transition: $custom-transition;
+      }
+    }
+  }
+
+  &.md-right {
+    flex-direction: row;
+
+    md-fab-trigger {
+      order: 1;
+    }
+
+    md-fab-actions {
+      flex-direction: row;
+      order: 2;
+
+      .md-fab-action-item {
+        transition: $custom-transition;
+      }
+    }
+  }
+
+  /*
+   * Handle the animations
+   */
+  &.scale {
+    .md-fab-action-item {
+      transform: scale(0);
+      transition: $custom-transition;
+    }
+  }
+
+  &.fling {
+    &.md-down {
+      .md-fab-action-item {
+        &:nth-child(1) { transform: translateY(-$trigger-height * 1 - $trigger-offset); z-index: 6; }
```

Comment:
Ah, thanks for the code snippet. I haven't seen the variable interpolation within SCSS before. I'll update this.